src/fields/fp/mod.rs

```diff
@@ -1063,6 +1064,22 @@ impl<'a, F: PrimeField> Sum<&'a FpVar<F>> for FpVar<F> {
     }
 }
 
+impl<'a, F: PrimeField> Sum<FpVar<F>> for FpVar<F> {
+    fn sum<I: Iterator<Item = FpVar<F>>>(iter: I) -> FpVar<F> {
+        let mut sum_constants = F::zero();
+        let sum_variables = FpVar::Var(AllocatedFp::<F>::addmany(iter.filter_map(|x| match x {
+            FpVar::Constant(c) => {
+                sum_constants += c;
+                None
+            },
+            FpVar::Var(v) => Some(v),
+        })));
+
+        let sum = sum_variables + sum_constants;
+        sum
+    }
+}
+
```

Comment:
Awesome! With this, we can directly call `sum` on an iterator of owned `FpVar`s, without the verbose `.collect().iter().sum()` (e.g., in https://github.com/privacy-scaling-explorations/sonobe/blob/f1d82418ba047cf90805f2d0505370246df24d68/folding-schemes/src/folding/circuits/nonnative/uint.rs#L692-L699) !